core/common/src/utils.rs

```diff
@@ -64,6 +76,105 @@ where
     last_error
 }
 
+/// Use Chariott to discover a service.
+///
+/// # Arguments
+/// * `chariott_uri` - Chariott's URI.
+/// * `namespace` - The service's namespace.
+/// * `name` - The service's name.
+/// * `version` - The service's version.
+/// # `communication_kind` - The service's communication kind.
+/// # `communication_reference` - The service's communication reference.
+pub async fn discover_service_using_chariott(
+    chariott_uri: &str,
+    namespace: &str,
+    name: &str,
+    version: &str,
+    communication_kind: &str,
+    communication_reference: &str,
+) -> Result<String, Status> {
+    let mut client = ServiceRegistryClient::connect(chariott_uri.to_string())
+        .await
+        .map_err(|e| Status::internal(e.to_string()))?;
+
+    let request = Request::new(DiscoverRequest {
+        namespace: namespace.to_string(),
+        name: name.to_string(),
+        version: version.to_string(),
+    });
+
+    let response = client.discover(request).await?;
+
+    let service = response.into_inner().service.ok_or_else(|| Status::not_found("Did not find a service in Chariott with namespace '{namespace}', name '{name}' and version {version}"))?;
+
+    if service.communication_kind != communication_kind
+        && service.communication_reference != communication_reference
+    {
+        return Err(Status::not_found(
+            "Did not find a service in Chariott with namespace '{namespace}', name '{name}' and version {version} that has communication kind '{communication_kind} and communication_reference '{communication_reference}''",
+        ));
+    }
+
+    Ok(service.uri)
+}
+
+/// Get a service's URI from settings or from Chariott.
+/// Will first try to use the URI defined in the service's settings file. If that is not set, will
+/// call Chariott to obtain it.
+///
+/// # Arguments
+/// * `service_uri` - Optional, desired service's URI.
+/// * `chariott_uri` - Optional, Chariott's URI.
+/// * `service_identifier` - Optional, The service's identifiers (name, namespace, version).
+/// # `communication_kind` - Optional, The service's communication kind.
+/// # `communication_reference` - Optional, The service's communication reference.
+pub async fn get_service_uri(
+    service_uri: Option<String>,
+    chariott_uri: Option<String>,
+    service_identifier: Option<ServiceIdentifier>,
+    communication_kind: &str,
```

Comment:
if you rename these in the above method don't forget to do it here too!

Reply:
changed